rpcs3/Emu/Cell/Modules/cellImeJp.cpp

```diff
@@ -43,10 +43,9 @@ bool ime_jp_manager::addChar(u16 c)
 	if (!c || cursor >= (CELL_IMEJP_STRING_MAXLENGTH - 1) || cursor > input_string.length())
 		return false;
 
-	std::u16string tmp;
-	tmp += c;
-	input_string.insert(cursor++, tmp);
-	cursor_end = cursor;
+	input_string += c; // resize
+	std::memmove(input_string.data() + cursor + 1, input_string.data() + cursor, sizeof(u16) * (input_string.size() - 1 - cursor));
```

Comment:
I don't wanna test this but this really looks weird.
From a quick glance this entire change seems to change the logic.

Reply:
It should work like before, I did workaround compiler bug.

Reply:
It's completely broken now